language/move-lang/src/typing/core.rs

```diff
@@ -817,6 +822,27 @@ pub fn make_function_type(
                 ])
             }
         }
+        FunctionVisibility::Friend(_) => {
+            if !in_current_module {
```

Comment:
Just for my own style when writing rules like this. I would probably do
```
FunctionVisibility::Friend(_) if in_current_module() || context.current_module_is_friend(m) => (),
FunctionVisibility::Friend(floc) => { error }
```

Reply:
I would also maybe try to change Script to that format as well. 

I'm not convinced what I'm saying is better, just that is how I would have started writing this match as I like to keep all the language rules as crisp and top level as possible in matches. It is why you will sometimes see me matching on a tuple of bools at various points

Reply:
I like the suggestion actually, especially given that the logic to decide whether it can be called is not that complicated. I'll update accordingly.

Reply:
Updated in 129744f